src/lib/list/list.scss

```diff
@@ -127,6 +127,31 @@ $mat-list-item-inset-divider-offset: 72px;
     }
   }
 
+  // If avatar is the 4th item, it means the list-item has both checkbox and avatar-icon.
+  // This means that there should be a margin for the mat-list-avatar on the start-side.
+  .mat-list-item-content .mat-list-avatar:nth-child(4) {
+    margin-right: 0;
+    margin-left: $mat-list-side-padding;
+
+    [dir='rtl'] & {
+      margin-right: $mat-list-side-padding;
+      margin-left: 0;
+    }
+  }
+
+  // Reversed content is mainly used by the MatSelectionList for displaying the checkbox at the
+  // end of the list option. If avatar is the 4th item, it means the list-item has both checkbox
+  // and avatar-icon. There needs to be a padding for the mat-list-avatar on the end-side.
+  .mat-list-item-content-reverse .mat-list-avatar:nth-child(4) {
```

Comment:
These selectors seem brittle. If the user has a `display: none` element somewhere inbetween, it'll throw everything off. You could work around it by setting a class on the avatar instead.

Reply:
Added padding to `.mat-list-text` instead and removed the `:not(:nth-child(2))` since it's hard to override.